Removed header file breaks compatibility for using codes

I got the following error when compiling ASPECT with the recent deal.II master:
```
In file included from /home/rene/software/aspect/source/boundary_composition/spherical_constant.cc:24:
/home/rene/software/aspect/include/aspect/geometry_model/spherical_shell.h:28:10: fatal error: 
      'deal.II/grid/tria_boundary_lib.h' file not found
#include <deal.II/grid/tria_boundary_lib.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
CMakeFiles/aspect.dir/build.make:278: recipe for target 'CMakeFiles/aspect.dir/source/boundary_composition/spherical_constant.cc.o' failed
```

The corresponding header in deal.II was removed in #6582. Of course I can remove the include from ASPECT and fix our development version, but our recent release will not work with deal.II master any more. Would it make sense to keep this header with a warning that it will be removed in a future release?

--- a/include/deal.II/grid/tria_boundary.h
+++ b/include/deal.II/grid/tria_boundary.h
@@ -1,0 +1,21 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_tria_boundary_h
+#define dealii_tria_boundary_h
+
+#warning This file is deprecated. Use the Manifold classes instead.
+
+#endif

--- a/include/deal.II/grid/tria_boundary_lib.h
+++ b/include/deal.II/grid/tria_boundary_lib.h
@@ -1,0 +1,21 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 1998 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_tria_boundary_lib_h
+#define dealii_tria_boundary_lib_h
+
+#warning This file is deprecated. Use the Manifold classes instead.
+
+#endif

